CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_VERSION 1)
+cmake_minimum_required(VERSION 3.11)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
+
+message("Platform: ${PLATFORM}")
+
+# Target-platform specific settings
+if("${PLATFORM}" STREQUAL "arm")
+    set(CMAKE_C_COMPILER_WORKS 1)
+    set(TOOLCHAIN_PREFIX arm-none-eabi-)
+    set(CMAKE_C_COMPILER ${TOOLCHAIN_PREFIX}gcc)
+    set(CMAKE_ASM_COMPILER ${TOOLCHAIN_PREFIX}gcc)
+    set(CMAKE_OBJCOPY ${TOOLCHAIN_PREFIX}objcopy)
+    set(CMAKE_OBJDUMP ${TOOLCHAIN_PREFIX}objdump)
+    set(SIZE ${TOOLCHAIN_PREFIX}size)
+elseif("${PLATFORM}" STREQUAL "x86")
+# We assume this cmake is running on a x86 platform, so the
+# default compiler settings should be fine
+else()
+    message(FATAL_ERROR
+        "ERROR: Please set the PLATFORM variable to 'x86' or 'arm'.")
+endif()
+
+# Setup project.
+if("${PLATFORM}" STREQUAL "arm")
+    project("UBC Formula Electric Software" C ASM)
+elseif("${PLATFORM}" STREQUAL "x86")
+    project("UBC Formula Electric Software" C CXX)
+endif()
+
+# Globally accessible paths
+set(REPO_ROOT_DIR ${CMAKE_SOURCE_DIR})
+set(BOARDS_DIR ${CMAKE_SOURCE_DIR}/boards)
+set(SCRIPTS_DIR ${CMAKE_SOURCE_DIR}/scripts)
+set(PIPENV_PROJECT_DIR ${REPO_ROOT_DIR})
+
+# Add option to disable invoking python from a virtual environment.
+# Invocation of python should be done with PYTHON_CMD.
+# Used in CI where packages are installed system-wide in a Docker container at build time.
+# This was done as a workaround, since I couldn't figure out how to pre-install packages into
+# a Pipenv virtual environment when building the container.
+option(NO_VENV "Disable invoking python scripts from virtual environment" OFF)
+
+if(NO_VENV)
+    set(PYTHON_CMD python3)
+else()
+    set(PYTHON_CMD pipenv run python)
+endif()
+
+# If compiling on x86, fetch GoogleTest for unit testing.
+if("${PLATFORM}" STREQUAL "x86")
+    include(FetchContent)
+    FetchContent_Declare(
+        googletest
+        URL https://github.com/google/googletest/archive/03597a01ee50ed33e9dfd640b249b4be3799d395.zip
+        DOWNLOAD_EXTRACT_TIMESTAMP 1
```

Comment:
What is this line for? I had to comment it out to be able to build on linux, see error below

``` 
/home/formulae/Documents/clion-2022.2.3/bin/cmake/linux/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DPLATFORM=x86 -G "CodeBlocks - Unix Makefiles" -S /home/formulae/Documents/test/Consolidated-Firmware -B /home/formulae/Documents/test/Consolidated-Firmware/cmake-build-gtest
Platform: x86
CMake Error at /home/formulae/Documents/clion-2022.2.3/bin/cmake/linux/share/cmake-3.23/Modules/ExternalProject.cmake:2670 (message):
  At least one entry of URL is a path (invalid in a list)
Call Stack (most recent call first):
  /home/formulae/Documents/clion-2022.2.3/bin/cmake/linux/share/cmake-3.23/Modules/ExternalProject.cmake:3640 (_ep_add_download_command)
  CMakeLists.txt:15 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!
See also "/home/formulae/Documents/test/Consolidated-Firmware/cmake-build-gtest/_deps/googletest-subbuild/CMakeFiles/CMakeOutput.log".

CMake Error at /home/formulae/Documents/clion-2022.2.3/bin/cmake/linux/share/cmake-3.23/Modules/FetchContent.cmake:1076 (message):
  CMake step for googletest failed: 1
Call Stack (most recent call first):
  /home/formulae/Documents/clion-2022.2.3/bin/cmake/linux/share/cmake-3.23/Modules/FetchContent.cmake:1217:EVAL:2 (__FetchContent_directPopulate)
  /home/formulae/Documents/clion-2022.2.3/bin/cmake/linux/share/cmake-3.23/Modules/FetchContent.cmake:1217 (cmake_language)
  /home/formulae/Documents/clion-2022.2.3/bin/cmake/linux/share/cmake-3.23/Modules/FetchContent.cmake:1260 (FetchContent_Populate)
  CMakeLists.txt:65 (FetchContent_MakeAvailable)


-- Configuring incomplete, errors occurred!
See also "/home/formulae/Documents/test/Consolidated-Firmware/cmake-build-gtest/CMakeFiles/CMakeOutput.log".

[Failed to reload]
```